Web: Add compact prop for ButtonText and fixes existing uses spacing

kimlisa opened the pull request:
enterprise: https://github.com/gravitational/teleport.e/pull/4855

fixes the weird spacings introduced with the new ButtonText

it's not the best, but better than being too far in the page (previous to the new button change, it was inline with other beginning elements)

an example:

| before | after |
| ------ | ----- |
|<img width="428" alt="image" src="https://github.com/user-attachments/assets/72748dc6-e647-4e07-ba81-b9245140562f"> |<img width="413" alt="image" src="https://github.com/user-attachments/assets/07200a2e-f0d6-42a2-86f8-c25639fd12bb">

kimlisa committed:
Add compact prop field for ButtonText

kimlisa committed:
Web: Fix ButtonText padding

kimlisa commented on web/packages/design/src/Button/Button.tsx:
```diff
@@ -409,5 +407,12 @@ export const ButtonWarningBorder = <E extends React.ElementType = 'button'>(
 export const ButtonText = <E extends React.ElementType = 'button'>(
-  props: ButtonProps<E>
-) => <Button fill="minimal" intent="neutral" {...props} />;
+  props: ButtonProps<E> & { compact?: boolean }
+) => (
+  <Button
+    fill="minimal"
+    intent="neutral"
+    {...(props.compact ? { px: 1 } : {})}
+    {...props}
+  />
+);
```
@bl-nero i took your compact suggestion, is this what you were envisioing?

bl-nero replied:
@kimlisa I was thinking about "same padding as with `inputAlignment` (which would be `px: 2`), unless `size=small`", but if an even smaller padding is commonly required, I have no objection here.

I would, however, declare the prop on the `Button` itself. My reason is that these additional components should be just wrappers, i.e. I should be equally well able to use `<Button fill="minimal" intent="neutral" compact/>` and achieve the same result.

kimlisa replied:
how about this: https://github.com/gravitational/teleport/pull/45241/commits/02f61df5dff1402a2eef2d8e7c36aee606911752#diff-a5c3729e6a710232f16bb4cbd373a966923e36fd2c6ade283ff0babef859f304R64

bl-nero replied:
LGTM

kimlisa committed:
Address CRs

kimlisa committed:
Use shouldForwardProp

Prevents non html attribute field "compact" from being
passed to dom which fails test
